src/main/java/org/owasp/html/HtmlStreamRenderer.java

```diff
@@ -252,23 +253,20 @@ private final void writeCloseTag(String uncanonElementName)
         StringBuilder cdataContent = pendingUnescaped;
         pendingUnescaped = null;
         Encoding.stripBannedCodeunits(cdataContent);
-        int problemIndex = checkHtmlCdataCloseable(lastTagOpened, cdataContent);
+        problemIndex = checkHtmlCdataCloseable(lastTagOpened, cdataContent);
```

Comment:
Let's take it either way. 
- No closing tag, not suspicious strings/tags skip it.
- No closing tag, suspicious string/tags break it. 
what says?

Reply:
> No closing tag, suspicious string/tags break it.
So when iteration runs to check open/close tags, it iterates over the entire individual HTML component. As of current if it doesn't get a closing tag, it breaks and the mail is empty. After this PR the particular invalid component is skipped and it's no more part of the HTML component hence we're safe from any JS injection issues if any in this block.
and any other scenarios of malicious code injection would be handled in the sanitization part.
As per me don't think to look for suspicious strings/tags. Anyway would like to know what exactly is suspicious string/tags

Reply:
I think you did not understand the code thoroughly. Let's connect over the call and walk through the code.